include/spdlog/sinks/daily_file_sink.h

```diff
@@ -59,7 +59,7 @@ struct daily_filename_format_calculator
         tm_format.push_back(' ');
 
         const size_t MIN_SIZE = 10;
-        filename_t buf;
+        std::vector<char> buf;
```

Comment:
```suggestion
        filename_t buf;
```

This line should not be changed.
`filename_t` can be an alias for `std::wstring`.

https://github.com/gabime/spdlog/blob/57a9fd0841f00e92b478a07fef62636d7be612a8/include/spdlog/common.h#L121-L129

If you care that `std::string::data()` returns `const char *`, replace `buf.data()` with `&buf[0]`.
See also: [Can I get a non\-const C string back from a C\+\+ string? \- Stack Overflow](https://stackoverflow.com/questions/1919626/can-i-get-a-non-const-c-string-back-from-a-c-string)

Reply:
It does not look nice, but unter the premise of wstring-support and that fmt's implementation uses &buf[0] as well, i'm okay with that.